Sources/PropertyEditor.swift

```diff
@@ -0,0 +1,680 @@
+import Result
+
+/// A mutable, observable property that has an optionally failable action
+/// associated with the setter.
+///
+/// ## Consistency when nested
+///
+/// `PropertyEditor` would back-propagate the validation failure if a
+/// proposed value from an outer property fails with its inner property.
+///
+/// It would also evaluate values originated from the inner property. In other
+/// words, it is possible for `value` to be invalid, causing `validations`
+/// to be a `failure`. Rely on `validations` for asserting a pass in validation.
+public final class PropertyEditor<Value, ValidationError: Error> {
+	/// The commited value of the property.
+	///
+	/// It does not guarantee that the value is valid with regard to `self`, since
+	/// changes might be initiated from the inner properties. Check `validations`
+	/// for the latest validation state.
+	public var commited: Property<Value>
+
+	/// The lifetime of the property.
+	public let lifetime: Lifetime
+
+	/// The latest validation that have been made by the property.
+	public let result: Property<ValidationResult<Value, ValidationError>>
```

Comment:
Either this is supposed to be called `validations`, or the comment for `commited` (sic) needs to be updated. I think it's probably just the latter.

Assuming it stays `result` I would suggest the comment reads:

/// The result of the last attempted edit on the underlying property

(Unless my understanding is broken, of course.)